### ISSUE TITLE
ZWave Thermostat mode classes use BigDecimal

### ISSUE THREAD

**Issue by cdjackson** [openhab/openhab#2174]
@digitaldan - do you know if there is there a reason that the thermostat mode classes use BigDecimal? I suspect it's because when they were written, the setpoint class used BigDecimal, so this was also used for the other classes.
I came across this when adding a converter elsewhere, but I don't think it's needed (8 bytes to represent 8 bits!) and have removed it (using Integer instead) as BigDecimal is expensive.
I'm going to push this into my "configuration item" branch since that's where I came across it - I was wondering if you could grab a copy and give it a quick test? (or if you think it's needed, please tell me I'm stupid! :) ).
I'll reference the commit here...

**Comment by digitaldan** [comment]
Well your not stupid, thats for sure ;-)  I have no idea why I used that,  since it was the first time I had touched the zwave code I probably followed the setpoint code as you mentioned. I'm pulling down your branch right now and let you know how it goes in a few.

**Comment by cdjackson** [comment]
Hold on with the I’m not stupid bit :) I think I forgot to add the extra classes into the repo - I renamed DecimalConverter to BigDecimal…  2 seconds and it should be there…

**Comment by digitaldan** [comment]
So I am having an issue with the Mode and Fan Mode values , they are showing as un-initialized in the sitemap.  I can see that the thermostats are sending the correct value and that we at least process them corretcly in the command classes, but I don't think the converter is working
```
2015-02-22 11:56:45.198 DEBUG o.o.b.z.i.p.s.ApplicationCommandMessageClass[:57]- NODE 7: Incoming command class THERMOSTAT_MODE (0x40)
2015-02-22 11:56:45.199 TRACE o.o.b.z.i.p.s.ApplicationCommandMessageClass[:80]- NODE 7: Found Command Class THERMOSTAT_MODE, passing to handleApplicationCommandRequest
2015-02-22 11:56:45.199 DEBUG o.o.b.z.i.p.c.ZWaveThermostatModeCommandClass[:95]- NODE 7: Received Thermostat Mode Request
2015-02-22 11:56:45.199 TRACE o.o.b.z.i.p.c.ZWaveThermostatModeCommandClass[:134]- NODE 7: Process Thermostat Mode Report
2015-02-22 11:56:45.199 DEBUG o.o.b.z.i.p.c.ZWaveThermostatModeCommandClass[:157]- NODE 7: Thermostat Mode report, value = 3
2015-02-22 11:56:45.200 DEBUG o.o.b.z.i.p.c.ZWaveThermostatModeCommandClass[:173]- NODE 7: Thermostat Mode Report, value = Auto
2015-02-22 11:56:45.200 DEBUG o.o.b.z.i.p.ZWaveController[:674]- Notifying event listeners
2015-02-22 11:56:45.200 TRACE o.o.b.z.i.p.ZWaveController[:677]- Notifying org.openhab.binding.zwave.internal.ZWaveActiveBinding@1c9d716
2015-02-22 11:56:45.200 DEBUG o.o.b.z.i.ZWaveActiveBinding[:433]- ZwaveIncomingEvent
2015-02-22 11:56:45.201 DEBUG o.o.b.z.i.ZWaveActiveBinding[:450]- NODE 7: Got a value event from Z-Wave network, endpoint = 0, command class = THERMOSTAT_MODE, value = 3
2015-02-22 11:56:45.201 TRACE o.o.b.z.i.c.ZWaveConverterHandler[:340]- Getting converter for item = Down_HVAC_HeatSetPoint, command class = THERMOSTAT_MODE, item command class = thermostat_setpoint
2015-02-22 11:56:45.201 TRACE o.o.b.z.i.c.ZWaveConverterHandler[:340]- Getting converter for item = Down_HVAC_Mode, command class = THERMOSTAT_MODE, item command class = thermostat_mode
2015-02-22 11:56:45.202 WARN  o.o.b.z.i.c.ZWaveThermostatModeConverter[:74]- NODE 7: No converter found for item = Down_HVAC_Mode, endpoint = 0, ignoring event.
2015-02-22 11:56:45.202 TRACE o.o.b.z.i.c.ZWaveConverterHandler[:340]- Getting converter for item = Down_HVAC_Operating_State, command class = THERMOSTAT_MODE, item command class = thermostat_operating_state
2015-02-22 11:56:45.202 TRACE o.o.b.z.i.c.ZWaveConverterHandler[:340]- Getting converter for item = Down_HVAC_CoolSetPoint, command class = THERMOSTAT_MODE, item command class = thermostat_setpoint
2015-02-22 11:56:45.202 TRACE o.o.b.z.i.c.ZWaveConverterHandler[:340]- Getting converter for item = Down_HVAC_Fan_Mode, command class = THERMOSTAT_MODE, item command class = thermostat_fan_mode
2015-02-22 11:56:45.203 TRACE o.o.b.z.i.c.ZWaveConverterHandler[:340]- Getting converter for item = Down_HVAC_Temperature, command class = THERMOSTAT_MODE, item command class = sensor_multilevel
2015-02-22 11:56:45.203 TRACE o.o.b.z.i.c.ZWaveConverterHandler[:340]- Getting converter for item = Down_HVAC_Fan_State, command class = THERMOSTAT_MODE, item command class = thermostat_fan_state 
```

**Comment by cdjackson** [comment]
Sorry - I missed the events… I’ll update in a couple of minutes…

Chris

**Comment by cdjackson** [comment]
Hopefully that sorts it…

**Comment by digitaldan** [comment]
Great, I give it another go in just a few

**Comment by digitaldan** [comment]
So Thermostat mode is now working, but the Fan Mode is still not.  Additionally Fan state does not appear to be working now (it was with the previous build)
```
2015-02-22 12:48:57.376 DEBUG o.o.b.z.i.p.s.ApplicationCommandMessageClass[:56]- NODE 7: Incoming command class THERMOSTAT_FAN_MODE
2015-02-22 12:48:57.377 TRACE o.o.b.z.i.p.s.ApplicationCommandMessageClass[:79]- NODE 7: Found Command Class THERMOSTAT_FAN_MODE, passing to handleApplicationCommandRequest
2015-02-22 12:48:57.377 DEBUG o.o.b.z.i.p.c.ZWaveThermostatFanModeCommandClass[:95]- NODE 7: Received Thermostat Fan Mode Request
2015-02-22 12:48:57.378 TRACE o.o.b.z.i.p.c.ZWaveThermostatFanModeCommandClass[:135]- NODE 7: Process Thermostat Fan Mode Report
2015-02-22 12:48:57.378 DEBUG o.o.b.z.i.p.c.ZWaveThermostatFanModeCommandClass[:158]- NODE 7: Thermostat Fan Mode report value = 0
2015-02-22 12:48:57.379 DEBUG o.o.b.z.i.p.c.ZWaveThermostatFanModeCommandClass[:172]- NODE 7: Thermostat Fan Mode Report value = Auto Low
2015-02-22 12:48:57.380 DEBUG o.o.b.z.i.p.ZWaveController[:674]- Notifying event listeners: ZWaveCommandClassValueEvent
2015-02-22 12:48:57.380 DEBUG o.o.b.z.i.ZWaveActiveBinding[:433]- ZwaveIncomingEvent
2015-02-22 12:48:57.381 DEBUG o.o.b.z.i.ZWaveActiveBinding[:450]- NODE 7: Got a value event from Z-Wave network, endpoint = 0, command class = THERMOSTAT_FAN_MODE, value = 0
2015-02-22 12:48:57.381 TRACE o.o.b.z.i.c.ZWaveConverterHandler[:340]- Getting converter for item = Down_HVAC_HeatSetPoint, command class = THERMOSTAT_FAN_MODE, item command class = thermostat_setpoint
2015-02-22 12:48:57.382 TRACE o.o.b.z.i.c.ZWaveConverterHandler[:340]- Getting converter for item = Down_HVAC_Mode, command class = THERMOSTAT_FAN_MODE, item command class = thermostat_mode
2015-02-22 12:48:57.383 TRACE o.o.b.z.i.c.ZWaveConverterHandler[:340]- Getting converter for item = Down_HVAC_Operating_State, command class = THERMOSTAT_FAN_MODE, item command class = thermostat_operating_state
2015-02-22 12:48:57.383 TRACE o.o.b.z.i.c.ZWaveConverterHandler[:340]- Getting converter for item = Down_HVAC_CoolSetPoint, command class = THERMOSTAT_FAN_MODE, item command class = thermostat_setpoint
2015-02-22 12:48:57.384 TRACE o.o.b.z.i.c.ZWaveConverterHandler[:340]- Getting converter for item = Down_HVAC_Fan_Mode, command class = THERMOSTAT_FAN_MODE, item command class = thermostat_fan_mode
2015-02-22 12:48:57.385 WARN  o.o.b.z.i.c.ZWaveThermostatFanModeConverter[:74]- NODE 7: No converter found for item = Down_HVAC_Fan_Mode, endpoint = 0, ignoring event.
2015-02-22 12:48:57.385 TRACE o.o.b.z.i.c.ZWaveConverterHandler[:340]- Getting converter for item = Down_HVAC_Temperature, command class = THERMOSTAT_FAN_MODE, item command class = sensor_multilevel
2015-02-22 12:48:57.386 TRACE o.o.b.z.i.c.ZWaveConverterHandler[:340]- Getting converter for item = Down_HVAC_Fan_State, command class = THERMOSTAT_FAN_MODE, item command class = thermostat_fan_state
```

```
2015-02-22 12:49:12.425 DEBUG o.o.b.z.i.p.s.ApplicationCommandMessageClass[:56]- NODE 45: Incoming command class THERMOSTAT_FAN_STATE
2015-02-22 12:49:12.425 TRACE o.o.b.z.i.p.s.ApplicationCommandMessageClass[:79]- NODE 45: Found Command Class THERMOSTAT_FAN_STATE, passing to handleApplicationCommandRequest
2015-02-22 12:49:12.426 DEBUG o.o.b.z.i.p.c.ZWaveThermostatFanStateCommandClass[:88]- NODE 45: Received Thermostat Fan State Request
2015-02-22 12:49:12.426 TRACE o.o.b.z.i.p.c.ZWaveThermostatFanStateCommandClass[:92]- NODE 45: Process Thermostat Fan State Report
2015-02-22 12:49:12.426 DEBUG o.o.b.z.i.p.c.ZWaveThermostatFanStateCommandClass[:115]- NODE 45: Thermostat fan state report value = 0
2015-02-22 12:49:12.426 DEBUG o.o.b.z.i.p.c.ZWaveThermostatFanStateCommandClass[:130]- NODE 45: Thermostat fan state  Report value = Idle
2015-02-22 12:49:12.426 DEBUG o.o.b.z.i.p.ZWaveController[:674]- Notifying event listeners: ZWaveCommandClassValueEvent
2015-02-22 12:49:12.427 DEBUG o.o.b.z.i.ZWaveActiveBinding[:433]- ZwaveIncomingEvent
2015-02-22 12:49:12.427 DEBUG o.o.b.z.i.ZWaveActiveBinding[:450]- NODE 45: Got a value event from Z-Wave network, endpoint = 0, command class = THERMOSTAT_FAN_STATE, value = 0
2015-02-22 12:49:12.427 TRACE o.o.b.z.i.c.ZWaveConverterHandler[:340]- Getting converter for item = Up_HVAC_Temperature, command class = THERMOSTAT_FAN_STATE, item command class = sensor_multilevel
2015-02-22 12:49:12.427 TRACE o.o.b.z.i.c.ZWaveConverterHandler[:340]- Getting converter for item = Up_HVAC_Fan_State, command class = THERMOSTAT_FAN_STATE, item command class = thermostat_fan_state
2015-02-22 12:49:12.428 WARN  o.o.b.z.i.c.ZWaveThermostatFanStateConverter[:71]- NODE 45: No converter found for item = Up_HVAC_Fan_State endpoint = 0, ignoring event.
2015-02-22 12:49:12.428 TRACE o.o.b.z.i.c.ZWaveConverterHandler[:340]- Getting converter for item = Up_HVAC_HeatSetPoint, command class = THERMOSTAT_FAN_STATE, item command class = thermostat_setpoint
2015-02-22 12:49:12.428 TRACE o.o.b.z.i.c.ZWaveConverterHandler[:340]- Getting converter for item = Up_HVAC_CoolSetPoint, command class = THERMOSTAT_FAN_STATE, item command class = thermostat_setpoint
2015-02-22 12:49:12.429 TRACE o.o.b.z.i.c.ZWaveConverterHandler[:340]- Getting converter for item = Up_HVAC_Fan_Mode, command class = THERMOSTAT_FAN_STATE, item command class = thermostat_fan_mode
2015-02-22 12:49:12.429 TRACE o.o.b.z.i.c.ZWaveConverterHandler[:340]- Getting converter for item = Up_HVAC_Mode, command class = THERMOSTAT_FAN_STATE, item command class = thermostat_mode
2015-02-22 12:49:12.429 TRACE o.o.b.z.i.c.ZWaveConverterHandler[:340]- Getting converter for item = Up_HVAC_Operating_State, command class = THERMOSTAT_FAN_STATE, item command class = thermostat_operating_state
```

**Comment by cdjackson** [comment]
I hope I’ve got them all this time. I guess with the fan that was working previously I probably removed BigDec on one side of the event, but not the other - hopefully (hopefully!) I’ve got them all now!

Thanks.

**Comment by digitaldan** [comment]
Everything looks good, let me know if you need any other testing!

**Comment by cdjackson** [comment]
Thanks Dan - appreciate it.

**Comment by cdjackson** [comment]
This change was tested by Dan and the change is now merged in #2198
